### PR TITLE
[WIP] Refactoring to support 2.0

### PR DESCRIPTION
**Pull request by WouterJ**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | #82, #83, #202
| License       | MIT
| Doc PR        | todo

While updating to 2.0, I also tried to make the code cleaner and easier to understand. Since this will be released as CmfMenuBundle 2.0, we can have BC breaks in it. This is kind of the last change to make things really better.

Todo
---

* [ ] Create functional tests for PublishWorkflow listeners
* [ ] Create custom Loader to support PublishWorkflow
* [x] Create a tag in the CoreBundle to support decorating services (which is support by Sf 2.5)
* [x] Update SE (https://github.com/symfony-cmf/standard-edition/pull/28)
* [ ] Update sandbox (https://github.com/symfony-cmf/cmf-sandbox/pull/284)
* [ ] Update cmf website
* [ ] Update docs
* [ ] Have working knp menu version (https://github.com/KnpLabs/KnpMenu/issues/198)

### CODE REVIEW

**Comment by dbu** [comment]
what is missing here?

**Comment by WouterJ** [comment]
@dbu some failing PRs, but mostly trouble with the sandbox and missing features in KnpMenu 2.0. I'm going to open a PR there soon and hope they'll merge it quick (and release a new minor version).

**Comment by dbu** [comment]
great. anything we can help with at this point?

**Comment by WouterJ** [comment]
I don't think so.

See https://github.com/KnpLabs/KnpMenu/pull/189 for the Knp PR.

**Comment by lsmith77** [comment]
needs a rebase

**Comment by coudenysj** [comment]
Any help needed?

**Comment by WouterJ** [comment]
No, this PR is almost ready. However, we're still waiting for some KnpMenu contributor to merge https://github.com/KnpLabs/KnpMenuBundle/pull/251

**Comment by WouterJ** [comment]
This PR is now finally passing.

The bundle itself is ready, but I haven't used it in a more complex application like the sandbox, because of missing features in the KnpMenuBundle 2.0 (for instance, determining if a menu item was current in a template). This will be changed in 2.1, but there are no release plans yet.

This wonders which direction we should take: Leave this in 2.0-dev/beta till KnpMenuBundle 2.1 is released and release a 1.3 release together with CMF 1.3 or release a 2.0 version of this bundle, knowing that it would be a bit harder to use it in projects.

**Comment by lsmith77** [comment]
doh .. the wait on KnpMenu never seems to stop :(

**Comment by dbu** [comment]
i think releasing a cmf menu 2.0 that has severe regressions is not a  good idea, so rather lets wait - or see if we can speed up things for knp menu 2.1 ourselves?

can you link all issues that are a problem for us, so that we know what to look at / when things would be ready?

**Comment by WouterJ** [comment]
@dbu I've created an issue in the KnpMenu repository to ask the maintainers about the status. I've pinged you about it.

I hope they (maybe with help of us) will be able to wrap things up soon. SonataAdminBundle 2.4 now requires KnpMenuBundle 2, so we have to merge this in order to use the new Sonata bundles.

File: QuietFactory.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class QuietFactory implements FactoryInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    private $innerFactory;
+    /**
```
Comment (lsmith77) [inline QuietFactory.php]: @WouterJ can you clean this up?

**Comment by lsmith77** [comment]
can you open follow up tickets for anything that still needs to be done?

**Comment by WouterJ** [comment]
will do

**Pull request merged**